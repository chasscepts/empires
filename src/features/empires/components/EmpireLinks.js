import PropTypes from 'prop-types';
import { Link } from 'react-router-dom';
import styles from '../css/EmpireLinks.module.css';
import Wrapper from './Wrapper';

function EmpireLink({ id, name }) {
  return (
    <Link to={`/empire/${id}`}>
      <div className={styles.link}><span>{name}</span></div>
    </Link>
  );
}

EmpireLink.propTypes = {
  id: PropTypes.number.isRequired,
  name: PropTypes.string.isRequired,
};

export default function EmpireLinks({ empires }) {
  return (
    <Wrapper>
      <h2 className={styles.heading}>Civilizations</h2>
      <ul className={styles.wrap}>
        {empires.map(
          (empire, index) => (
            <li key={empire.id} className={styles.square}>
              <EmpireLink id={empire.id} name={empire.name} index={index} />
            </li>
          ),
        )}
      </ul>
    </Wrapper>
  );
}

EmpireLinks.propTypes = {
  empires: PropTypes.arrayOf(
    PropTypes.shape({
      id: PropTypes.number.isRequired,
      name: PropTypes.string.isRequired,
      expansion: PropTypes.string.isRequired,
      army_type: PropTypes.string.isRequired,
      unique_unit: PropTypes.arrayOf(PropTypes.string).isRequired,
      unique_tech: PropTypes.arrayOf(PropTypes.string).isRequired,
      team_bonus: PropTypes.string.isRequired,
      civilization_bonus: PropTypes.arrayOf(PropTypes.string).isRequired,
    }),
  ).isRequired,
};
