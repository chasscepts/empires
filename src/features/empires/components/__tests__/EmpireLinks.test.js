import React from 'react';
import renderer from 'react-test-renderer';
import { render, screen } from '@testing-library/react';
import sample from '../test_helpers/sampleEmpires';
import RouterWrapper from '../test_helpers/RouterWrapper';
import EmpireLinks from '../EmpireLinks';

jest.mock('axios');

function Links () {
  return <EmpireLinks empires={sample.civilizations} />;
}

describe('EmpireLinks', () => {
  it('matches snapshot', () => {
    const tree = renderer
      .create(<RouterWrapper component={Links} />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('displays all empire links', () => {
    render(<RouterWrapper component={Links} />);
    expect(screen.getByText(sample.civilizations[0].name)).toBeInTheDocument();
    expect(screen.getByText(sample.civilizations[1].name)).toBeInTheDocument();
  });
});
