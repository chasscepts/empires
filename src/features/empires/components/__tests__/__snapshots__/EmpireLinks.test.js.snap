// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EmpireLinks matches snapshot 1`] = `
<div
  className="container"
>
  <header
    className="header"
  >
    <h1>
      Age Of Empires II
    </h1>
    <nav>
      <a
        className="link"
        href="/"
        onClick={[Function]}
        role="link"
      >
        üè†
      </a>
    </nav>
  </header>
  <h2
    className="heading"
  >
    Civilizations
  </h2>
  <ul
    className="wrap"
  >
    <li
      className="square"
    >
      <a
        href="/empire/1"
        onClick={[Function]}
      >
        <div
          className="link"
        >
          <span>
            Aztecs
          </span>
        </div>
      </a>
    </li>
    <li
      className="square"
    >
      <a
        href="/empire/2"
        onClick={[Function]}
      >
        <div
          className="link"
        >
          <span>
            Britons
          </span>
        </div>
      </a>
    </li>
  </ul>
</div>
`;
